cljfmt/src/cljfmt/core.cljc

```diff
@@ -145,14 +145,32 @@
 (defn- comment-next? [zloc]
   (-> zloc z/next* skip-whitespace comment?))
 
-(defn- should-indent? [zloc]
-  (and (line-break? zloc) (not (comment-next? zloc))))
-
-(defn- should-unindent? [zloc]
-  (and (indentation? zloc) (not (comment-next? zloc))))
-
-(defn unindent [form]
-  (transform form edit-all should-unindent? z/remove*))
+;; Distinguishes line comments (;;), which may be indentable,
+;; from heading comments (;;;) and inline comments (;), which are not
+(defn- comment-next-other-than-line-comment? [zloc]
+  (when-let [znext (-> zloc z/next* skip-whitespace)]
+    (when (comment? znext)
+      (let [text (str (z/node znext))]
+        (or (not (str/starts-with? text ";;"))
+            (str/starts-with? text ";;;"))))))
```

Comment:
Could split out the `line-comment?` predicate? That way we have code that clearly and precisely defines what a line comment is. So something like:

```clojure
(defn- line-comment? [zloc]
  (and (comment? zloc) (re-matches #";;[^;].*" (z/string zloc))))

(defn- comment-next-other-than-line-comment? [zloc]
  (when-let [znext (-> zloc z/next* skip-whitespace)]
    (and (comment? znext) (not (line-comment? znext)))))
```

Reply:
Perfect, thanks!